### PR TITLE
[icmp] add `OT_ICMP6_ECHO_HANDLER_RLOC_ALOC_ONLY` to `otIcmp6EchoMode`

### PR DESCRIPTION
**Pull request by sunytt**
When an icmp echo request is sent to rloc/aloc address of a Thread device, OT replies to the packet. The packet is also passed to host and the kernel may reply to the echo request. This would cause 2 echo replies and may be confusing when testing packet loss rate by pinging a rloc/aloc address.

Added OT_ICMP6_ECHO_HANDLER_RLOC_ALOC_ONLY to otIcmp6EchoMode to let OT only handle echo request to rloc/aloc addresses, this is the default mode in posix code. Set otIcmp6EchoMode to OT_ICMP6_ECHO_HANDLER_DISABLED will help to avoid double ping replies from both OT and host.

### CODE REVIEW

File: src/core/config/ip6.h [hunk 1]
```diff
@@ -237,6 +237,16 @@
...
 
+/**
+ * @def OPENTHREAD_CONFIG_IP6_HANDLE_ECHO_REQUEST_DESTINED_FOR_LOCATOR
+ *
+ * Define as 1 to handle ICMP echo request destined for locator addresses.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_IP6_HANDLE_ECHO_REQUEST_DESTINED_FOR_LOCATOR
+#define OPENTHREAD_CONFIG_IP6_HANDLE_ECHO_REQUEST_DESTINED_FOR_LOCATOR 1
+#endif
```
Comment (abtink) [inline src/core/config/ip6.h:248]: The change in this PR LGTM. Thanks. 👍 

Did a bit of background, looks like the change to `// always handle Echo Request destined for RLOC or ALOC` was added way back in https://github.com/openthread/openthread/pull/3011.

One idea regarding this:
- We have `otIcmp6EchoMode` which allows users to set the behavior regarding handling of echo requests by OT stack.
- We can consider adding the new behavior (handling pings for RLOC/ALOC) as a new option in this `enum`?
- This way, we have one way of configuring how pings are handled.

Maybe the following:

```c++
typedef enum otIcmp6EchoMode
{
    OT_ICMP6_ECHO_HANDLER_DISABLED       = 0, ///< Disabled
    OT_ICMP6_ECHO_HANDLER_UNICAST_ONLY   = 1, ///< Enabled only for unicast requests only
    OT_ICMP6_ECHO_HANDLER_MULTICAST_ONLY = 2, ///< Enabled only for multicast requests only
    OT_ICMP6_ECHO_HANDLER_ALL            = 3, ///< Enabled for unicast and multicast requests
    OT_ICMP6_ECHO_HANDLER_RLOC_ALOC_ONLY = 4, ///< Enabled for RLOC/ALOC destinations only
} otIcmp6EchoMode;
```

Thoughts?

**Review by abtink — APPROVED** [review]

File: src/core/config/ip6.h (hunk 1 above)
Comment (sunytt) [reply src/core/config/ip6.h:248]: Thanks for the comment, checked with @librasungirl that #3011 was for wpantund. Do you know if wpantund should continue to be supported @jwhui ? or can we simply remove the special handling for rloc/aloc in #3011?

Comment (sunytt) [reply src/core/config/ip6.h]: Thanks @abtink for the suggestion, I updated the code to switch from using new flag to using new enum in `otIcmp6EchoMode`. The posix code sets the mode to `OT_ICMP6_ECHO_HANDLER_RLOC_ALOC_ONLY` to keep the existing behavior, wpantund would be supported if it's used by anyone. For systems that want to disable echo handling from OT completely they can set the mode to `OT_ICMP6_ECHO_HANDLER_DISABLED`.

**Comment by sunytt** [comment]
> Given we are already mapping all addresses with host IP stack, does disabling ICMP echo solves the problem?

Offline discussed, disabling ICMP echo completely will solve the double ICMP echo replies problem.

**Review by abtink — APPROVED** [review]
Looks great! Thank you. 👍

File: src/posix/platform/netif.cpp [hunk 2]
```diff
@@ -2250,7 +2250,7 @@ void platformNetifSetUp(void)
 #if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
     otIcmp6SetEchoMode(gInstance, OT_ICMP6_ECHO_HANDLER_ALL);
 #else
-    otIcmp6SetEchoMode(gInstance, OT_ICMP6_ECHO_HANDLER_DISABLED);
+    otIcmp6SetEchoMode(gInstance, OT_ICMP6_ECHO_HANDLER_RLOC_ALOC_ONLY);
```
Comment (bukepo) [inline src/posix/platform/netif.cpp:2253]: Does linux need this change? I think this is only required for NCP build. Probably set this in `src/ncp/ncp_base_mtd`.

Comment (sunytt) [reply src/posix/platform/netif.cpp]: linux doesn't need `OT_ICMP6_ECHO_HANDLER_RLOC_ALOC_ONLY`, moved this to `ncp_base`, could you take another look?

**Review by bukepo — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**